CMakeLists.txt

```diff
@@ -697,20 +697,23 @@ if(gnutls OR NOT OPENSSL_FOUND)
 	elseif(gnutls)
 		message(FATAL_ERROR "GnuTLS library not found")
 	endif()
-endif()
-
-if(NOT OPENSSL_FOUND)
-	find_public_dependency(LibGcrypt)
-	set_package_properties(LibGcrypt
-		PROPERTIES
-			URL "https://www.gnupg.org/software/libgcrypt/index.html"
-			DESCRIPTION "A general purpose cryptographic library"
-			TYPE RECOMMENDED
-			PURPOSE "Use GCrypt instead of the built-in functions for RC4 and SHA1"
-	)
-	if (LibGcrypt_FOUND)
-		target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_LIBGCRYPT)
-		target_link_libraries(torrent-rasterbar PRIVATE LibGcrypt::LibGcrypt)
+else()
```

Comment:
this changed indentation, so you may want to ignore white space changes

Reply:
Currently you have:

```
if(NOT gnutls)
# find and enable openssl....

....

if(gnutls OR NOT TARGET OpenSSL::SSL)
# find and enable gnutls, fail if not found
```

which means that if you pass `-Dgnutls=OFF` and `OpenSSL` is not found inside the first if condition, the second if condition will be satisfied (due to `OR NOT TARGET OpenSSL::SSL`), and thus `gnutls` will be silently used, or the build will fail with `GnuTLS library not found`, which is probably not expected.

Reply:
so, are you suggesting, instead of an `else()` block here, restore the `if (NOT OPENSSL_FOUND)` below, but change it to `if (NOT TARGET OpenSSL AND NOT gnutls)`?

Reply:
@arvidn 

I think the problem is that the current code cannot distinguish between the user passing `-Dgnutls=OFF` explicitly and simply not being defined, which has the same effect.

I would expect that if the user passes `-Dgnutls=OFF` explicitly, they would expect the build system:
- to use `libssl` if it exists (which is currently the case)
- to fail if they their system has `gnutls` but not `libssl` (currently not the case - `gntls` is going to be used regardless)

So I think it should be:

```
if(NOT gnutls)
# find and enable OpenSSL if available. note that if it's available, it will always be used
else()
# find and enable gnutls, fail if not found
endif()
```

In the future, you may want to make the conditons of find/use of OpenSSL more explicit (for example, failing the build unless some kind of feature like `NO_TLS_LIBRARY` is passed).

Reply:
I would not expect the second bullet point. I would expect `gnutls=OFF` to mean "build without using gnutls" (i.e. not failing). The semantics are very confusing given that all other options, other than gnutls, are implicit. So, from the point of view that the implicit linking against openssl is reasonable, saying `gnutls=OFF` would still make openssl eligible to be linked against, but not required.

I think all these options should be only explicit, but I think that should be a separate patch, definitely not changed in a merge commit.